### PR TITLE
Feat: Simple note taking app for Particular video

### PR DESCRIPTION
**Pull request by pritam-kr**
- [Feat: Notes app in video library](https://github.com/pritam-kr/playme/commit/7b31f0224ecabb31ef4cce8b621009d9a7531c24)
- [Helper function for scroll to top](https://github.com/pritam-kr/playme/commit/410d63b50779231b2dc300468735c7cd0aa711d0)
- [Separated private route](https://github.com/pritam-kr/playme/commit/1a7175ccc8b88552a36d0f544aaf263d8a365cb4)
- Delete notes for particular video

Screenshot (64)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -5,52 +5,46 @@ import { Explore, Login, Playlist, SingleVideo,Signup, WatchLater , Likes, Histo
...
         <Route path="/likes" element={<PrivateRoute><Likes /></PrivateRoute>} > </Route>
         <Route path="/watchlater" element={<PrivateRoute><WatchLater /></PrivateRoute>} > </Route>
-
         <Route path="/*" element={<Navigate  to="/explore" />}></Route> 
-
         <Route path="/playlist/:playlistId" element={<PrivateRoute><PlaylistVideos /></PrivateRoute>} > </Route>
         <Route path="/latest" element={<SortByLatest />} ></Route>
         <Route path="/profile" element={<PrivateRoute><Profile /></PrivateRoute>} ></Route>
       </Routes>
+      </Scroll>
```
Comment (poornima-alugubelly) [inline src/App.js:47]: You can use replace as tanay taught yesterday

File: src/Hooks/Notes.js [hunk 2]
```diff
@@ -1,31 +1,28 @@
...
+    const {dispatch} = useVideoContext()
+     const createNotes = async (note, token, videoId) => {
 
         try{
-            const response = await axios.post("api/user/notes", {note}, {headers: {authorization: token}})
+            const {data: {video}, status} = await axios.post(`/api/video/${videoId}`, {note}, {headers: {authorization: token}})
+
+             if(status === 201){
+                dispatch({type: "ADD_NOTE", payload: video})
+                toast.success("Notes Added", {position: "top-right"})
```
Comment (poornima-alugubelly) [inline src/Hooks/Notes.js]: Can we put the contants in a  constants file?

File: src/Utils/GetCategories.js [hunk 3]
```diff
@@ -1,5 +1,6 @@
 export const uniqueCategory = (data, category) => {
-  const categoryName = data.map((eachProduct) => eachProduct[category]);
+
+  const categoryName = data?.map((eachProduct) => eachProduct[category]);
```
Comment (poornima-alugubelly) [inline src/Utils/GetCategories.js]: I think this was supposed to be videos instead of product

**Review by poornima-alugubelly — APPROVED** [review]
Good work! Everything is working as expected

File: src/Hooks/Notes.js (hunk 2 above)
Comment (pritam-kr) [reply src/Hooks/Notes.js]: Yes, I'll do refactor.

> Good work! Everything is working as expected

Thank you Poornima

File: src/Utils/GetCategories.js (hunk 3 above)
Comment (pritam-kr) [reply src/Utils/GetCategories.js]: Done

File: src/App.js (hunk 1 above)
Comment (pritam-kr) [reply src/App.js:47]: Ok noted

**Review by preethasri — APPROVED** [review]
UI looks awesome and the notes feature too .great work. just a suggestion is notes are adding even if it is empty .u can make it required* for those inputs

**Review by MaruthiPothuganti — COMMENTED** [review]
Well written code and Everything looks good. Just an addition though- You can make the input fields of notes app blank, once a note is created.

**Comment by pritam-kr** [comment]
> UI looks awesome and the notes feature too .great work. just a suggestion is notes are adding even if it is empty .u can make it required* for those inputs

Done

**Pull request merged**